Dockerfile.multiarch

```diff
@@ -36,47 +49,64 @@ RUN --mount=type=cache,target=/root/.cache/go-build \
 #
 FROM base AS builder
 
+ARG TARGETARCH
+WORKDIR /go/src/github.com/ssvlabs/ssv/
+
 # Copy source code
 COPY . .
 
-# Set up compiler based on target architecture
-RUN if [ "$TARGETARCH" = "amd64" ]; then \
-      export CC=x86_64-linux-gnu-gcc; \
-    elif [ "$TARGETARCH" = "arm64" ]; then \
-      export CC=aarch64-linux-gnu-gcc; \
-    elif [ "$TARGETARCH" = "arm" ]; then \
-      export CC=arm-linux-gnueabihf-gcc; \
-    fi && \
+# Pull in target glibc static libs for -static to find libm.a and libmvec.a
+RUN set -eux; \
+    TARGET_DEB_ARCH=$(case "$TARGETARCH" in \
+      amd64) echo amd64 ;; \
+      arm64) echo arm64 ;; \
+      *)    echo armhf ;; \
+    esac); \
+    dpkg --add-architecture "$TARGET_DEB_ARCH"; \
+    apt-get update; \
+    apt-get install -y --no-install-recommends libc6-dev:"$TARGET_DEB_ARCH"; \
+    rm -rf /var/lib/apt/lists/*
+
+# Build the static binary
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,mode=0755,target=/go/pkg \
     COMMIT=$(git rev-parse HEAD) && \
     VERSION=$(git describe --tags $(git rev-list --tags --max-count=1) --always) && \
-    echo "Building version: $VERSION (commit: $COMMIT)" && \
-    CGO_ENABLED=1 GOOS=linux GOARCH=$TARGETARCH \
-    go install \
-      -tags="blst_enabled" \
-      -ldflags "-X main.Commit=$COMMIT -X main.Version=$VERSION -linkmode external -extldflags '-static -lm'" \
-      ./cmd/ssvnode
+    CC=$(case "$TARGETARCH" in \
+      amd64) echo x86_64-linux-gnu-gcc ;; \
+      arm64) echo aarch64-linux-gnu-gcc ;; \
+      *)    echo arm-linux-gnueabihf-gcc ;; \
+    esac) && \
+    echo "Building version $VERSION (commit $COMMIT) for $TARGETARCH" && \
+    CGO_ENABLED=1 GOOS=linux GOARCH=$TARGETARCH CC=$CC \
+      go build -tags="blst_enabled" \
+        -ldflags "\
+          -X main.Commit=$COMMIT \
+          -X main.Version=$VERSION \
+          -s -w \
+          -linkmode external \
+          -extldflags '-static -lm'" \
+        -o /go/bin/ssvnode ./cmd/ssvnode
 
 #
-# STEP 3: Prepare image to run the binary
+# STEP 3: Runner image
 #
-FROM golang:1.24 AS runner
+FROM --platform=$TARGETPLATFORM golang:1.24 AS runner
 
 # Install minimal runtime dependencies
 RUN apt-get update && \
-  DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
-  dnsutils && \
-  rm -rf /var/lib/apt/lists/*
+    DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends \
+      dnsutils && \
+    rm -rf /var/lib/apt/lists/*
 
 # Copy binary and configuration
 WORKDIR /
 COPY --from=builder /go/bin/ssvnode /go/bin/ssvnode
-COPY ./Makefile .env* ./
+COPY Makefile .env* ./
 COPY config/* ./config/
 
-# Expose port for load balancing
+# Expose ports and set DNS resolver
 EXPOSE 5678 5000 4000/udp
-
-# Force using Go's DNS resolver because Alpine's DNS resolver (when netdns=cgo) may cause issues.
```

Comment:
Do we need to delete this?

Reply:
The functionality is still there, I merged comments. And the comment was incorrect and outdated, we are using debian based Go, not Alpine

Reply:
More info here: https://github.com/ssvlabs/ssv/issues/2192